### PR TITLE
Colors and interactions for user cards

### PR DESCRIPTION
**Pull request by MykolaGalian**
Added colors and interactions for grades and content views user cards (only for test data).

Functional:
Local App work as expected with test data.
Browsers (Windows 10 Chrome, Firefox, Edge)

Func tests: 
  - Click on a series of marks it selected. All other courses are greyed
  - Legend reflects selected courses properly
  - Click on a selected series unselect it
  - Once all series are deselected then the course filter is cleared. All series deselected = show all series

Accessibility: n/a
Security: n/a
Devops: n/a
UI: in progress

FYI @johngwilkinson @anhill-D2L @bemailloux

### CODE REVIEW

**Comment by NicholasHallman** [comment]
Hmmm I need to work on how the legend interprets it's interactions. Atm this chart works like the charts in the main dashboard where selecting the first item "focuses" it and later changes "toggle" until the selected array is empty again. The legend is simpler, all clicks are toggles. If the legend is translucent then it means that course is disabled, not focused as is the behavior here. I'm going to bring it up in the slack thread vitali made and hopefully we'll get a more conclusive answer

File: components/content-views-card.js [hunk 1]
```diff
@@ -173,25 +193,52 @@ class ContentViewsCard extends SkeletonMixin(Localizer(MobxLitElement)) {
...
+	_orgUnitName(orgUnitId) {
+		const orgUnit = this._serverData.orgUnits.find(unit => unit[ORG_UNIT.ID] === orgUnitId);
+
+		return orgUnit ? orgUnit[ORG_UNIT.NAME] : '';
+	}
+
+	get _series() {
+		if (!this.data._data) return [];
+
+		const colors = Array.from({ length: this._trendData.length }, function() { return this.next().value; }, UserTrendColorsIterator(0, 1, this._trendData.length));
```
Comment (NicholasHallman) [inline components/content-views-card.js:233]: this line can be replaced with

```javascript
const colors = [...UserTrendColorsIterator(0, 1, this._trendData.length)]
```

**Comment by Vitalii-Misechko** [comment]
@MykolaGalian 
Could you double-check if you can activate a course with keyboard for `Content View Over Time` and `Grades Over Time`
Could you make pascal case for `Content view over time` title?

File: components/grades-trend-card.js [hunk 2]
```diff
@@ -173,25 +193,52 @@ class GradesTrendCard extends SkeletonMixin(Localizer(MobxLitElement)) {
...
+	_orgUnitName(orgUnitId) {
+		const orgUnit = this._serverData.orgUnits.find(unit => unit[ORG_UNIT.ID] === orgUnitId);
+
+		return orgUnit ? orgUnit[ORG_UNIT.NAME] : '';
+	}
+
+	get _series() {
+		if (!this.data._data) return [];
+
+		const colors = Array.from({ length: this._trendData.length }, function() { return this.next().value; }, UserTrendColorsIterator(0, 1, this._trendData.length));
```
Comment (NicholasHallman) [inline components/grades-trend-card.js:233]: here too ^

File: components/content-views-card.js (hunk 1 above)
Comment (Vitalii-Misechko) [reply components/content-views-card.js:233]: nice!

File: test/components/user-drill-view.test.js [hunk 3]
```diff
@@ -214,7 +222,7 @@ describe('d2l-insights-user-drill-view', () => {
 		it('should render the alert if there are >= 10 courses', async() => {
 			data.recordsByUser = new Map();
 			data.recordsByUser.set(232, []);
-			(new Array(11)).fill([]).forEach((record, i) => data.recordsByUser.get(232).push([i, 232]));
+			(new Array(11)).fill([]).forEach((record, i) => data.recordsByUser.get(232).push([i + 1, 232]));
```
Comment (NicholasHallman) [inline test/components/user-drill-view.test.js:225]: Good catch thanks!

**Review by NicholasHallman — APPROVED** [review]

File: components/access-trend-card.js [hunk 4]
```diff
@@ -226,7 +226,7 @@ class AccessTrendCard extends SkeletonMixin(Localizer(MobxLitElement)) {
 				{ x: Date.UTC(2020, 2, 21), y: 6 }
 			]
 		}, {
-			orgUnitId: 3,
+			orgUnitId: 8,
```
Comment (Vitalii-Misechko) [inline components/access-trend-card.js:229]: Cool! There is  orgUnit with ID: 8

File: test/components/content-views-card.test.js [hunk 5]
```diff
@@ -22,7 +31,24 @@ describe('content-views-card', () => {
...
+
+	it('should grey out all other courses when a first course is selected', async() => {
+		const selectedCourses = new SelectedCourses();
+		const el = await fixture(html`<d2l-insights-content-views-card .data="${data}" .selectedCourses="${selectedCourses}"></d2l-insights-content-views-card>`);
+		const series = el.shadowRoot.querySelector('d2l-labs-chart').chart.series;
+
+		el._toggleFilterEventHandler(series[0]);
+		await el.updateComplete;
+
+		expect(el._series[0].color).to.deep.equal('#4885DC');
```
Comment (Vitalii-Misechko) [inline test/components/content-views-card.test.js:50]: `nit` .`deep.equal` is overkill here. It seems that `equal` or `eql` covers all needs.

File: test/components/grades-trend-card.test.js [hunk 6]
```diff
@@ -23,6 +32,23 @@ describe('grades-trend-card', () => {
...
+		const selectedCourses = new SelectedCourses();
+		const el = await fixture(html`<d2l-insights-grades-trend-card .data="${data}" .selectedCourses="${selectedCourses}"></d2l-insights-grades-trend-card>`);
+		const series = el.shadowRoot.querySelector('d2l-labs-chart').chart.series;
+
+		el._toggleFilterEventHandler(series[0]);
+		await el.updateComplete;
+
+		expect(el._series[0].color).to.deep.equal('#4885DC');
+		expect(el._series[1].color).to.deep.equal(gray);
+		expect(el._series[2].color).to.deep.equal(gray);
```
Comment (Vitalii-Misechko) [inline test/components/grades-trend-card.test.js:52]: same

**Comment by Vitalii-Misechko** [comment]
Could you add into functional section cases that were tested?

**Review by Vitalii-Misechko — APPROVED** [review]

**Pull request merged**